app/controllers/api/base_controller/generic.rb

```diff
@@ -28,6 +29,12 @@ def add_resource(type, _id, data)
         end
       end
 
+      def validate_type(klass, type)
+        klass.descendant_get(type)
+      rescue => err
```

Comment:
Being as narrow as possible when rescuing exceptions is good, and since we know exactly the error we're expecting in this scenario, let's catch that specific one - `rescue ArgumentError => err`